api/src/main/java/com/spotify/metrics/core/RemoteSemanticMetricRegistry.java

```diff
@@ -104,4 +104,9 @@
      * @return a new {@link RemoteMeter}
      */
     RemoteMeter meter(final MetricId name);
+
+    /**
+     * Wait for all metrics to be sent to the remote.
+     */
+    void waitForAllCalls();
```

Comment:
I'll make it a shutdown method with a timeout and a flag preventing additional submissions.